### PR TITLE
Finished cesium load error.

### PR DESCRIPTION
**Pull request by teamagam3**
Cesium now reloads every time the map failes to load.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/49" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java [hunk 1]
```diff
@@ -264,6 +270,15 @@ public boolean isReady() {
         return mIsGGMapReadySynchronized.getData();
     }
 
+    @Override
+    public void onError(String error) {
+        if(!mIsHandlingError.getData()) {
+            mIsHandlingError.setData(true);
+            init();
+            loadUrl(FILE_ANDROID_ASSET_VIEWER);
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: why do you load the url second time?
you have the same line at the end of init !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    CesiumJsErrorListener mErrorListener;
+
+    public CesiumWebChromeClient(CesiumJsErrorListener errorListener) {
+        mErrorListener = errorListener;
+    }
+
+
+
+    @Override
+    public boolean onConsoleMessage(ConsoleMessage consoleMessage) {
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java:22]: are you sure that consoleMessage is not null and has all the fields?

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        // Notify that the onConsoleMessage is overridden
+        return true;
+    }
+
+    private boolean shouldRaiseEvent(String fileSource, String message) {
+        // If the console message comes from Cesium, it's an error.
+        // A listener should be called.
+        if (fileSource.toLowerCase().contains(CESIUM_LIBRARY_FILE)) {
+            if (message.toLowerCase().contains(CESIUM_ERROR_MESSAGE)) {
+                return true;
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java:39]: that's the message? no details?

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java [hunk 4]
```diff
@@ -264,6 +270,15 @@ public boolean isReady() {
         return mIsGGMapReadySynchronized.getData();
     }
 
+    @Override
+    public void onError(String error) {
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: give it more detailed name. !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java [hunk 5]
```diff
@@ -264,6 +270,15 @@ public boolean isReady() {
...
+        if(!mIsHandlingError.getData()) {
+            mIsHandlingError.setData(true);
+            init();
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: this event-loop is very heavy and it also run in the uithread which is not acceptable.

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java [hunk 6]
```diff
@@ -273,10 +288,15 @@ public boolean isReady() {
...
+
+
+
             //Runs runnable on UI thread
             CesiumMapView.this.post(new Runnable() {
                 @Override
                 public void run() {
+                    CesiumMapView.this.mIsHandlingError.setData(false);
+
                     CesiumMapView.this.mIsGGMapReadySynchronized.setData(true);
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:371]: can you use mIsGGMapReadySynchronized for this? this is not critical...

**Comment by teamagam1** [comment]
check

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java [hunk 7]
```diff
@@ -28,7 +28,9 @@
 /**
  * Wrapper view class for a WebView-based Cesium viewer
  */
-public class CesiumMapView extends WebView implements GGMapView, VectorLayer.LayerChangedListener {
+public class CesiumMapView
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:36]: why why why???

# @

[](url)

`public static void main(){
}`

> if you want to shoot shoot dont talk
- [x] complete
- [x] resolve
- דכגדכ

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java [hunk 8]
```diff
@@ -28,7 +28,9 @@
...
+        extends WebView
+        implements GGMapView, VectorLayer.LayerChangedListener, CesiumWebChromeClient.CesiumJsErrorListener {
 
     public static final String FILE_ANDROID_ASSET_VIEWER =
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:41]: clean

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java (hunk 7 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:36]: !completed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java (hunk 8 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:41]: !completed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java (hunk 4 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java (hunk 5 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java (hunk 1 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: !resolved

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java (hunk 6 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:371]: !resolved  !resolved

**Comment by teamagam1** [comment]
@agamgil יאללה

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumBaseBridge.java [hunk 9]
```diff
@@ -18,6 +18,8 @@ public CesiumBaseBridge(JavascriptCommandExecutor javascriptCommandExecutor) {
         mJsExecutor = javascriptCommandExecutor;
     }
 
+
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumBaseBridge.java]: no need for spalce, delete

Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumBaseBridge.java]: igone it

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+
+
+    @Override
+    public boolean onConsoleMessage(ConsoleMessage consoleMessage) {
+        String fileSource = consoleMessage.sourceId();
+        String message = consoleMessage.message();
+
+        if (shouldRaiseEvent(fileSource, message)) {
+            RaiseEvent(message);
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java]: camelCase !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java [hunk 11]
```diff
@@ -273,6 +302,21 @@ public boolean isReady() {
     }
 
     @Override
+    public void onCesiumError(String error) {
+        if (error.contains("ImageryProvider")) {
+            mIsHandlingError.setData(true);
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:298]: arent you suppose to insert data? why do we update with TRUE?

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java (hunk 2 above)
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java:22]: whats is the answer? what about try catch?

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java (hunk 10 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java (hunk 11 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:298]: כי עכשיו אנחנו יודעים שיש בעיה. אז אנחנו זוכרים את זה במידע

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java (hunk 2 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java:22]: !completed

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java:22]: yes we are sure

**Pull request merged**